### PR TITLE
Use centralised `assert.sh` script [DI-171]

### PR DESCRIPTION
**Pull request by JackPGreen**
The `assert.sh` script was [sourced externally](https://github.com/torokmark/assert.sh) but a local copy was included in the repository.

This change removes the local copy, instead dynamically fetching from a central [Hazelcast fork](https://github.com/hazelcast/assert.sh).

Our local copy had some small changes, which have been lost, but the previous version has been migrated to a branch of the Hazelcast fork to allow [a diff of the script](https://github.com/hazelcast/assert.sh/compare/main...hazelcast-packaging).

Specifically, the changes are only to improve formatting of assertion messages, not the assertions themselves.

Fixes: [DI-171](https://hazelcast.atlassian.net/browse/DI-171)

[DI-171]: https://hazelcast.atlassian.net/browse/DI-171?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JackPGreen** [comment]
@ldziedziul @nishaatr are you able to take a look at this, please?

**Review by nishaatr — APPROVED** [review]
LGTM at at back of https://github.com/hazelcast/hazelcast-docker/pull/779

**Comment by ldziedziul** [comment]
Waiting with approval update of logging in tests

**Comment by JackPGreen** [comment]
> Waiting with approval update of logging in tests

Done.

File: packages/brew/test_brew_functions.sh [hunk 1]
```diff
@@ -11,7 +14,8 @@ function assertAlphanumCamelCase {
   local testValue=$1
   local expected=$2
   local actual=$(alphanumCamelCase "$testValue")
-  assert_eq "$expected" "$actual" "Alphanumeric camel case of $testValue should be equal to $expected " || TESTS_RESULT=$?
+  local msg="Alphanumeric camel case of $testValue should be equal to $expected"
+  assert_eq "$expected" "$actual" "$msg" && log_success "$MSG" || TESTS_RESULT=$?
```
Comment (ldziedziul) [inline packages/brew/test_brew_functions.sh:18]: Casing is mixed up here and this file is not run during be `test_scripts.sh`. Could you rename this file `brew_functions_test.sh` to be picked up ?

Comment (JackPGreen) [reply packages/brew/test_brew_functions.sh]: > Casing is mixed up here

I'll fix that

8c8c7ae7ebdcbacb859c67fbfbad554400496fc9

> and this file is not run during be `test_scripts.sh`. Could you rename this file `brew_functions_test.sh` to be picked up ?

I've tried that, but unfortunately the script doesn't pass on `master` either. So for now I'll leave it alone, but raise as a separate issue for the future.

Comment (JackPGreen) [reply packages/brew/test_brew_functions.sh]: ...actually this might be a Mac + `sed` issue. Let me take another look.

Comment (JackPGreen) [reply packages/brew/test_brew_functions.sh]: 8547cd09fca299714bca3d38bd97693af1d60ffa

Renaming them works on non-Mac platforms.

**Comment by ldziedziul** [comment]
There's some issue here: https://github.com/hazelcast/hazelcast-packaging/actions/runs/10187951924/job/28207888467#step:4:36
image

**Comment by JackPGreen** [comment]
> There's some issue here: https://github.com/hazelcast/hazelcast-packaging/actions/runs/10187951924/job/28207888467#step:4:36 image

The action was running on the merge commit, as that test is new and wasn't in this branch... fixed.

8f66e417d186ea89570ce09d56d26c50266d5404

File: .github/workflows/build.functions_tests.sh [hunk 2]
```diff
@@ -53,7 +58,8 @@ function assert_get_hz_dist_tar_gz {
   local distribution=$2
   local expected_url=$3
   local actual_url=$(get_hz_dist_tar_gz "$hz_version" "$distribution")
-  assert_eq "$expected_url" "$actual_url" "Expected URL for version \"$hz_version\", distribution \"$distribution\"" || TESTS_RESULT=$?
+  local msg="Expected URL for version \"$hz_version\", distribution \"$distribution\""
```
Comment (ldziedziul) [inline .github/workflows/build.functions_tests.sh:61]: Could you add `$expected_url` to `$msg` to see what exactly we expect, currently we see only inputs:

image

Comment (JackPGreen) [reply .github/workflows/build.functions_tests.sh]: 70ad43a96fc5014e0f3bdd49f1b495a2dd8b6484

**Review by ldziedziul — APPROVED** [review]

**Pull request merged**